exist-core/src/test/xquery/unicode.xqm

```diff
@@ -70,6 +70,14 @@ function uni:sip-dom-persistent() {
         fn:string-to-codepoints(fn:doc($stored)/container/smp)
 };
 
+declare
+    %test:assertEquals(172998)
+function uni:sip-dom-persistent-text() {
+    let $stored := xmldb:store("/db/" || $uni:COLLECTION_NAME, "sip.xml", $uni:SIP)
+    return
+        fn:string-to-codepoints(fn:doc($stored/text())/container/smp)
```

Comment:
The `/text()` should be removed. `$stored` returns an `xs:string`, so a `/text()` node test would raise an error.

Reply:
The test should actually pass if you remove the `/text()`, at least I'm getting the expected codepoint when trying it in an isolated query.

Reply:
Same here - removing `text()` or moving it to the end of the path expression—i.e., `smp/text()`—allows the tests to pass:

```xml
<testsuite package="http://exist-db.org/xquery/test/unicode"
    timestamp="2021-01-04T14:34:33.964-05:00" tests="6" failures="0" errors="0" pending="0"
    time="PT0.01S">
    <testcase name="sip-dom-memtree" class="uni:sip-dom-memtree"/>
    <testcase name="sip-dom-memtree-transform" class="uni:sip-dom-memtree-transform"/>
    <testcase name="sip-dom-persistent" class="uni:sip-dom-persistent"/>
    <testcase name="sip-dom-persistent-text" class="uni:sip-dom-persistent-text"/>
    <testcase name="sip-dom-persistent-transform" class="uni:sip-dom-persistent-transform"/>
    <testcase name="sip-dom-persistent-transform-2" class="uni:sip-dom-persistent-transform-2"/>
</testsuite>
```